tsrc/workspace/linker.py

```diff
@@ -0,0 +1,54 @@
+from typing import List
+
+import cli_ui as ui
+from path import Path
+
+import tsrc
+import tsrc.executor
+import os
+
+
+class FileLinker(tsrc.executor.Task[tsrc.Link]):
+    def __init__(self, workspace_path: Path, repos: List[tsrc.Repo]) -> None:
+        self.workspace_path = workspace_path
+        self.repos = repos
+
+    def on_start(self, *, num_items: int) -> None:
+        ui.info_2("Creating symlinks")
+
+    def on_failure(self, *, num_errors: int) -> None:
+        ui.error("Failed to create the following symlinks:")
+
+    def display_item(self, item: tsrc.Link) -> str:
+        return f"{item.source} linking to {item.target}"
+
+    def process(self, index: int, count: int, item: tsrc.Link) -> None:
+        ui.info_count(index, count, "Linking", item.source, "->", item.target)
```

Comment:
I think this works but there are two problems with this:

* You should put all operations in the try/except (to avoid tsrc sync crash)
* The code would be easier to read if it was split into smaller functions

Reply:
I have an idea for restructuring the code.  First, though, for this and the previous item I need to go read about try-except! :)

Reply:
Please give the latest commit a review.  I changed how errors get handled, just raising exceptions directly.  This yields both nicer output in the terminal and gets tsrc _not_ to say "Done" at the end. :)

I don't know how clean the diff view of the update will be, so I encourage you to examine the new code directly.